plots_and_notebooks/run_metacal_mof.ipynb

```diff
@@ -0,0 +1,297 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "import seaborn as sns\n",
+    "\n",
+    "%matplotlib notebook\n",
+    "sns.set()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import galsim\n",
+    "\n",
+    "from mdetsims import Sim\n",
+    "from metadetect.detect import MEDSifier\n",
+    "from mdetsims.defaults import TEST_METACAL_MOF_CONFIG\n",
+    "from mdetsims.metacal import get_fofs, MOFFitter, MetacalFitter\n",
+    "from mdetsims.metacal.util import get_masked_frac"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "des_sim = Sim(\n",
+    "    rng=np.random.RandomState(seed=100),\n",
+    "    g1=0.02,\n",
+    "    gal_type='wldeblend',\n",
+    "    scale=0.263,\n",
+    "    psf_type='gauss',\n",
+    "    n_coadd=30,\n",
+    "    n_coadd_psf=1,\n",
+    "    gal_kws={'survey_name': 'DES'})"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mbobs = des_sim.get_mbobs()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mfier = MEDSifier(\n",
+    "    mbobs=mbobs,\n",
+    "    sx_config=TEST_METACAL_MOF_CONFIG['sx'],\n",
+    "    meds_config=TEST_METACAL_MOF_CONFIG['meds'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "mbmeds = mfier.get_multiband_meds()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Try the FoFs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "res = get_fofs(mfier.cat, TEST_METACAL_MOF_CONFIG['fofs'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_fofs = np.max(res[1]['fofid'])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for fofid in range(n_fofs):\n",
+    "    msk = res[1]['fofid'] == fofid\n",
+    "    if np.sum(msk) > 2:\n",
+    "        inds = res[1]['number'][msk] - 1\n",
+    "        list_of_mbobs = []\n",
+    "        for ind in inds:\n",
+    "            o = mbmeds.get_mbobs(ind)\n",
+    "            o[0].meta['Tsky'] = 1\n",
+    "            o.meta['masked_frac'] = get_masked_frac(o)\n",
+    "            o[0].meta['magzp_ref'] = 26.5\n",
+    "            o.meta['id'] = ind + 1\n",
+    "            o.meta['fofid'] = fofid\n",
+    "            list_of_mbobs.append(o)"
```

Comment:
I've had to set some extra metadata on the input mbobs' in order to get the MOF code to run.

Reply:
that's true, I was not testing multi-epoch

Reply:
sorry that was to answer the other comment not this one